### PR TITLE
Skeletons

### PR DESCRIPTION
**Pull request by savelichalex**

### CODE REVIEW

File: packages/flask/src/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    useSharedValue,
+    runOnJS,
+    withSpring,
+} from 'react-native-reanimated';
+import LinearGradient from 'react-native-linear-gradient';
+
+import { useTheme, ColorVariants, UIBackgroundView } from '@tonlabs/uikit.hydrogen';
+import { SkeletonProgress, useSkeletonProgress } from './useSkeletonProgress';
+
+enum CrossDisolveProgress {
```
Comment (vetkamak) [inline packages/flask/src/Skeleton/Skeleton.tsx]:
```suggestion
enum CrossDissolveProgress {
```

File: packages/flask/src/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    Hidden = 0,
+}
+
+/**
+ * A *cross dissolve* is a post-production video editing technique
+ * in which you gently increase the opacity of one scene over the previous one.
+ */
+function useCrossDissolve(visible: boolean) {
+    const crossDisolveProgress = useSharedValue(
+        visible ? CrossDisolveProgress.Visible : CrossDisolveProgress.Hidden,
```
Comment (vetkamak) [inline packages/flask/src/Skeleton/Skeleton.tsx]:
```suggestion
        visible ? CrossDissolveProgress.Visible : CrossDissolveProgress.Hidden,
```

File: packages/flask/src/Skeleton/useSkeletonProgress.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    progress = makeMutable(SkeletonProgress.Start);
+
+    private activeIDs = new Set();
+
+    show(id: number) {
+        const wasInactive = this.activeIDs.size === 0;
+        this.activeIDs.add(id);
+
+        if (wasInactive) {
+            this.progress.value = withRepeat(
```
Comment (sertony) [inline packages/flask/src/Skeleton/useSkeletonProgress.tsx:24]: I already forgot that skeleton should have a common "progress" through all the skeletons ;)

**Review by sertony — APPROVED** [review]
I like the way you've implemented the shimmer effect!

File: packages/flask/src/Skeleton/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    Visible = 1,
+    Hidden = 0,
+}
+
+/**
+ * A *cross dissolve* is a post-production video editing technique
+ * in which you gently increase the opacity of one scene over the previous one.
+ */
+function useCrossDissolve(visible: boolean) {
+    const crossDisolveProgress = useSharedValue(
```
Comment (vetkamak) [inline packages/flask/src/Skeleton/Skeleton.tsx]: And one more 🙂
```suggestion
    const crossDissolveProgress = useSharedValue(
```

**Review by vetkamak — APPROVED** [review]

**Pull request merged**